.github/workflows/e2e_nightly.yml

```diff
@@ -0,0 +1,121 @@
+name: E2E Nightly
+
+# Remove before merge!
+on: [push, pull_request]
+  # Uncomment before merge!
+  # schedule:
+  # Run every day at 23:00 UTC
+  # - cron: '0 23 * * *'
+
+jobs:
+  adapter-build-hw:
+    name: Build SYCL, UR, run E2E
+    strategy:
+      matrix:
+        adapter: [
+          {name: CUDA}
+        ]
+        build_type: [Release]
+        compiler: [{c: clang, cxx: clang++}]
+
+    runs-on: ${{matrix.adapter.name}}
+
+    steps:
+    - name: Checkout UR
+      uses: actions/checkout@v4
+      with:
+        path: ur-repo
+
+    - name: Checkout SYCL
+      uses: actions/checkout@v4
+      with:
+        repository: intel/llvm
+        ref: sycl
+        path: sycl-repo
+
+    - name: Install pip packages
+      working-directory: ${{github.workspace}}/ur-repo
+      run: pip install -r third_party/requirements.txt
+
+    - name: Download DPC++
+      run: |
+        wget -O dpcpp_compiler.tar.gz https://github.com/intel/llvm/releases/download/nightly-2023-09-21/sycl_linux.tar.gz
+        mkdir dpcpp_compiler
+        tar -xvf dpcpp_compiler.tar.gz -C dpcpp_compiler
+
+    - name: Configure CMake UR
+      working-directory: ${{github.workspace}}/ur-repo
+      run: >
+        cmake
+        -B build
+        -DCMAKE_C_COMPILER=${{matrix.compiler.c}}
+        -DCMAKE_CXX_COMPILER=${{matrix.compiler.cxx}}
+        -DCMAKE_BUILD_TYPE=${{matrix.build_type}}
+        -DUR_ENABLE_TRACING=ON
+        -DUR_DEVELOPER_MODE=ON
```

Comment:
is "developer mode" needed here...?

Reply:
Enables "treat warnings as errors". I think we should leave it. Especially if we start using this code on PR.